utils/nvme.go

```diff
@@ -141,161 +140,113 @@ func (n *Nvme) cmdListCapabilities(ctx context.Context, logicalPath string) ([]b
+	parsedCaps, err = parseSanicap(caps.SANICAP)
+	if err != nil {
+		return nil, err
+	}
+	capabilitiesFound = append(capabilitiesFound, parsedCaps...)
 
-			// Generate short flag identifier
-			for _, word := range strings.Fields(data[1]) {
-				capability.Name += strings.ToLower(word[0:1])
-			}
+	return capabilitiesFound, nil
+}
 
-			capability.Description = data[1]
+func parseFna(fna uint) []*common.Capability {
+	// Bit masks values came from nvme-cli repo
+	// All names come from internal nvme-cli names
+	// We will *not* keep in sync as these names form our API
+	// https: // github.com/linux-nvme/nvme-cli/blob/v2.8/nvme-print-stdout.c#L2199-L2217
+
+	return []*common.Capability{
+		{
+			Name:        "fmns",
+			Description: "Format Applies to All/Single Namespace(s) (t:All, f:Single)",
+			Enabled:     (fna&(0b1<<0))>>0 != 0,
+		},
+		{
+			Name:        "cens",
+			Description: "Crypto Erase Applies to All/Single Namespace(s) (t:All, f:Single)",
+			Enabled:     (fna&(0b1<<1))>>1 != 0,
+		},
+		{
+			Name:        "cese",
+			Description: "Crypto Erase Supported as part of Secure Erase",
+			Enabled:     (fna&(0b1<<2))>>2 != 0,
+		},
+	}
+}
 
-			if enabled != "0" {
-				capability.Enabled = true
-			}
+func parseSanicap(sanicap uint) ([]*common.Capability, error) {
+	// Bit masks values came from nvme-cli repo
+	// All names come from internal nvme-cli names
+	// We will *not* keep in sync as these names form our API
+	// https://github.com/linux-nvme/nvme-cli/blob/v2.8/nvme-print-stdout.c#L2064-L2093
+
+	caps := []*common.Capability{
+		{
+			Name:        "cer",
+			Description: "Crypto Erase Sanitize Operation Supported",
+			Enabled:     (sanicap&(0b1<<0))>>0 != 0,
+		},
+		{
+			Name:        "ber",
+			Description: "Block Erase Sanitize Operation Supported",
+			Enabled:     (sanicap&(0b1<<1))>>1 != 0,
+		},
+		{
+			Name:        "owr",
+			Description: "Overwrite Sanitize Operation Supported",
+			Enabled:     (sanicap&(0b1<<2))>>2 != 0,
+		},
+		{
+			Name:        "ndi",
+			Description: "No-Deallocate After Sanitize bit in Sanitize command Supported",
+			Enabled:     (sanicap&(0b1<<29))>>29 != 0,
+		},
+	}
 
-			capability.Description = data[1]
-			capabilitiesFound = append(capabilitiesFound, capability)
-		}
+	switch (sanicap & (0b11 << 30)) >> 30 {
+	case 0b00:
+		// nvme prints this out for 0b00:
+		//   "Additional media modification after sanitize operation completes successfully is not defined"
+		// So I'm taking "not defined" literally since we can't really represent 2 bits in a bool
+		// If we ever want this as a bool we could maybe call it "dmmas" maybe?
+	case 0b01, 0b10:
+		caps = append(caps, &common.Capability{
+			Name:        "nodmmas",
+			Description: "Media is additionally modified after sanitize operation completes successfully",
+			Enabled:     (sanicap&(0b11<<30))>>30 == 0b10,
+		})
+	case 0b11:
+		return nil, errSanicapNODMMASReserved
+	default:
+		panic("unreachable")
```

Comment:
can we return an error here as well, instead of a panic - if you think code could end up here?

Reply:
Code should never reach down here, I think I might have done this only for linters but now I'm not sure, let me see if I can drop it.

Reply:
removed the panic, linters did not complain

Reply:
this panic ended up in main, looks like I messed up a rebase somewhere and undid some of my fixups to your comments :(, will fix these up for real.

Reply:
or maybe I never pushed them?